### PR TITLE
major refactoring in ApnsConnectingImpl to separate MonitoringThread fro...

### PR DESCRIPTION
**Pull request by bruffos**
...m current socket

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.notnoop.apns</groupId>
     <artifactId>apns</artifactId>
-    <version>1.0.0.Beta5-SNAPSHOT</version>
+    <version>1.0.0.bruffos.Beta5</version>
```
Comment (matzew) [inline pom.xml]: can you revert these changes ? and the other ones on the piom.xml ?

**Comment by matzew** [comment]
any updates on this, @froh42, @bruffos ?

File: pom.xml (hunk 1 above)
Comment (bruffos) [inline pom.xml]: I will revert that.
/P

**Comment by turf00** [comment]
While it is good that there are to now be separate cachedNotification queues on a per connection basis, I do not see how this fix addresses the race condition identified in  #193.  

Both the sendMessage thread and the monitor socket thread will still share the same ConnectionBean, therefore it will still be a race to see if sendMessage can add the notification to the cache before the monitor thread checks if the notification is in the cache.

While this is probably not likely when not running a simulator, GC or other thread scheduling could have this effect and therefore it probably will happen at some point.

I can see a few ways round this and might be able to spend some time next week creating a solution.

Also I pulled this code to my local machine and broke it by doing a longer number of runs at ApnsSimulatorLongRunningTest.multipleTokensBad_issue145

It seems that this test was modified to leave more time between the runs but that does not address all of the problems with the simulator.
